python/cudf/cudf/core/frame.py

```diff
@@ -1473,7 +1488,10 @@ def _drop_na_columns(self, how="any", subset=None, thresh=None):
                 thresh = len(df)
 
         for col in self._data.names:
-            if (len(df[col]) - df[col].null_count) < thresh:
+            if ((len(df[col]) - df[col].null_count) < thresh) or (
+                isinstance(df._data[col], cudf.core.column.NumericalColumn)
+                and ((len(df[col]) - df._data[col].nan_count) < thresh)
+            ):
```

Comment:
Let's encapsulate what it means for this condition to be true, as it's not obvious from looking at the code:


```python
something = ((len(df[col]) - df[col].null_count) < thresh) or (
                isinstance(df._data[col], cudf.core.column.NumericalColumn)
                and ((len(df[col]) - df._data[col].nan_count) < thresh)
            )

if something:
    ...
```

Reply:
Sure, addressed this